src/services/state-persistor/state-persistor.model.ts

```diff
@@ -20,6 +22,7 @@ export const SP_KEY_SRS = 'SRS' // TODO:
 export const SP_KEY_X = 'X'
 export const SP_KEY_Y = 'Y'
 export const SP_KEY_SERIAL = 'serial'
+export const SP_KEY_SERIAL_LAYERS = 'serialLayer'
```

Comment:
does this mean, serialLayer gets written to localStorage also? If I remember correclty, it should only be in the URL.

Reply:
State persistor does not necessarily mean keeping in both url and localstorage, but since you did not use the new state persistor mecanism at all, we must deep dive into the code and find where the process is made. Actually I did not modify the behavior of what you made, I have just replaced 

```
const serialLayer = new URLSearchParams(window.location.search).get(
      'serialLayer'
)
```

by 

```
const serialLayer = urlStorage.getItem(SP_KEY_SERIAL_LAYERS)
```